managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpFindNewest.java

```diff
@@ -92,10 +94,14 @@ public void readEntryComplete(Entry entry, Object ctx) {
                 return;
             } else {
                 lastMatchedPosition = position;
-
                 // check last entry
                 state = State.checkLast;
+                PositionImpl lastPosition = ledger.getLastPosition();
                 searchPosition = ledger.getPositionAfterN(searchPosition, max, PositionBound.startExcluded);
+                if (lastPosition.compareTo(searchPosition) < 0) {
+                    log.debug("first position {} matches, last should be {}, but moving to lastPos {}", position, searchPosition, lastPosition);
```

Comment:
add this to a `if (log.isDebugEnabled())` block.

Reply:
done